moabb/datasets/schirrmeister2017.py

```diff
@@ -77,182 +73,26 @@ def data_path(
-            clab_set = h5file["nfo"]["clab"][:].squeeze()
-            all_sensor_names = [
-                "".join(chr(c.squeeze()) for c in h5file[obj_ref]) for obj_ref in clab_set
-            ]
-            if pattern is not None:
-                all_sensor_names = filter(
-                    lambda sname: re.search(pattern, sname), all_sensor_names
-                )
-        return all_sensor_names
-
-    def _add_markers(self, cnt):
-        with h5py.File(self.filename, "r") as h5file:
-            event_times_in_ms = h5file["mrk"]["time"][:].squeeze()
-            event_classes = h5file["mrk"]["event"]["desc"][:].squeeze().astype(np.int64)
-
-            # Check whether class names known and correct order
-            # class_name_set = h5file['nfo']['className'][:].squeeze()
-            # all_class_names = [''.join(chr(c) for c in h5file[obj_ref])
-            #                    for obj_ref in class_name_set]
-
-        event_times_in_samples = event_times_in_ms * cnt.info["sfreq"] / 1000.0
-        event_times_in_samples = np.uint32(np.round(event_times_in_samples))
-
-        # Check if there are markers at the same time
-        previous_i_sample = -1
-        for i_event, (i_sample, _) in enumerate(
-            zip(event_times_in_samples, event_classes)
-        ):
-            if i_sample == previous_i_sample:
-                info = "{:d}: ({:.0f} and {:.0f}).\n".format(
-                    i_sample, event_classes[i_event - 1], event_classes[i_event]
-                )
-                log.warning(
-                    f"Same sample has at least two markers.\n{info}Marker codes will be summed."
-                )
-            previous_i_sample = i_sample
-
-        # Now create stim chan
-        stim_chan = np.zeros_like(cnt.get_data()[0])
-        for i_sample, id_class in zip(event_times_in_samples, event_classes):
-            stim_chan[i_sample] += id_class
-        info = mne.create_info(
-            ch_names=["STI 014"], sfreq=cnt.info["sfreq"], ch_types=["stim"]
-        )
-        stim_cnt = mne.io.RawArray(stim_chan[None], info, verbose="WARNING")
-        cnt = cnt.add_channels([stim_cnt])
-        event_arr = [
-            event_times_in_samples,
-            [0] * len(event_times_in_samples),
-            event_classes,
+        train_raw, test_raw = [
+            mne.io.read_raw_edf(path, infer_types=True)
+            for path in self.data_path(subject)
         ]
-        cnt.info["events"] = [
-            dict(list=np.array(event_arr).T, channels=None),
+
+        # Select only EEG sensors (remove EOG, EMG),
+        # and also set montage for visualizations
+        montage = make_standard_montage("standard_1005")
+        train_raw, test_raw = [
+            raw.pick_types(eeg=True).set_montage(montage) for raw in (train_raw, test_raw)
         ]
-        return cnt
+        sessions = {
+            "session_1": {"train": train_raw, "test": test_raw},
```

Comment:
```suggestion
            "session_0": {"run_0": train_raw, "run_1": test_raw},
```
We are trying to use consistent name for session and run number, i.e. using integer and starting from 0. Is that ok to rename the dict file as suggested?

Reply:
I think in general might be nice to keep the "train"/"test" for backward compatibility of all the code. Also run_0 and run_1 might be misleading as train is actually the concatenation of typically 11 runs and test typically the concatenation of 2 runs.

Reply:
I think could be nice to keep it train/test to keep backwards compatibility with code that did it this way? Also for this dataset, train was not one run, but typically 11 consecutive runs concatenated, and test 2 such consecutive runs. What do you think?

Reply:
I agree that in this case, we could keep the name "train" and "test".  If we need to access specific run in the future, we could open a PR for adding a parameter to the dataset class to specify that we want to load each run (without train/test name).

So, let's merge!